modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -526,24 +537,45 @@ private CompletableFuture<?> createZonePartitionReplicationNode(
                 rebalanceRetryDelayConfiguration
         );
 
+        var safeTimeTracker = new SafeTimeValuesTracker(HybridTimestamp.MIN_VALUE);
```

Comment:
Shall we move all this stuff inside the `Supplier` as well?

Reply:
Why?

Reply:
Why not? Its only used inside it